regulation-worker/internal/delete/api/api.go

```diff
@@ -41,9 +42,7 @@ func (*APIManager) GetSupportedDestinations() []string {
 	return supportedDestinations
 }
 
-// prepares payload based on (job,destDetail) & make an API call to transformer.
-// gets (status, failure_reason) which is converted to appropriate model.Error & returned to caller.
-func (api *APIManager) Delete(ctx context.Context, job model.Job, destination model.Destination) model.JobStatus {
+func (api *APIManager) deleteWithRetry(ctx context.Context, job model.Job, destination model.Destination, retryAttempts int) model.JobStatus {
```

Comment:
retryAttempts -> retryAttempt  or retryAttemptCount or something like makes more sense as it is indicating, up till how many attempts happened right?

Reply:
It can go either way:
- count how many retries happened until now
- current retry number

I'd prefer the latter.